components/gameplayIceGrid.tsx

```diff
@@ -0,0 +1,62 @@
+import { useEffect, useState } from 'react'
+import type { Hexagon } from '../types'
+import { IceGrid } from './iceGrid'
+import { isEqual } from '../utils/isEqual'
+import { getNeighbors } from '../utils/getNeighbors'
+
+
+type GameplayIceGridProps = {
+  hexagonList: Hexagon[]; /* Array of Tuples */
+  maxMistakes: number;
+  onLoose: () => void;
+  onWin: () => void;
+  path: Hexagon[];
+  startHexagons: Hexagon[];
+}
+
+export function GameplayIceGrid({
+  hexagonList,
+  maxMistakes,
+  onLoose,
+  onWin,
+  path,
+  startHexagons,
+}: GameplayIceGridProps) {
+  const [pathPosition, setPathPosition] = useState(0);
+  const [mistakeHexagons, setMistakeHexagons] = useState<Hexagon[]>([]);
+
+  const isWon = pathPosition === path.length;
+  const isLost = mistakeHexagons.length >= maxMistakes;
+  useEffect(() => {
+    if (isWon) {
+       onWin(); 
+    }
+  }, [isWon, onWin]);
+
+  useEffect(() => {
+    if (isLost) {
+      onLoose();
+    }
+  }, [isLost, onLoose]);
```

Comment:
Maybe we have to move this state into the parent component after all. Let's keep it in mind for some other time.